_posts/2022-03-16-solidity-0.8.13-release-announcement.md

```diff
@@ -0,0 +1,206 @@
+Please read more about it in the [security alert](https://blog.soliditylang.org/2022/03/16/encodecall-bug/).
+
+## Notable New Features
+
+### Yul IR Pipeline Production Ready
+
+We have been working on Yul as an intermediate language for Solidity for several years now.
+Yul in itself has been considered production ready for a while - essentially since
+ABI coder V2 was marked "stable", since it makes heavy use of Yul. The other part,
+code generation from Solidity to Yul, can now also be considered stable and production ready.
+
+It has been feature complete for several months and since the beginning, we always ran all our
+tests through both compiler pipelines. Over the previous weeks, we added several more complex
+real-world contracts to our testing infrastructure and fixed the last bugs in the component.
+
+You can enable the new pipeline using the ``--via-ir`` commandline switch or the setting
+``{"viaIR": true}`` (at the same sevel where you would have ``"optimizer": true``).
+
+It is not recommended to use the Yul compiler pipeline without optimizer because we wrote the
+intermediate code in an extremely modular and maintainable way that assumes that any excess code
+is optimized away.
+
+The performance of the new pipeline is not yet always superior to the old one, but it
+can do much higher-level optimization across functions, so please try it out and give us feedback!
+
+One important feature of the new pipeline is that it can move stack variables to memory
+and thus avoid the "stack too deep" issue. In order to do this safely, the compiler needs to know
+that the memory slots it intends to use are not overwritten by inline assembly.
+Because of that, inline assembly should only use memory that has been previously allocated
+either by high-level Solidity code or by reading from the free memory pointer at ``0x40``.
+At the very beginning of the code, the compiler will set the free memory pointer to a position
+after the area reserved for the stack variables that were moved to memory.
+Since it is difficult to impossible to check by the compiler that inline assembly only accesses memory in a safe way,
+we added a syntactic mechanism for the developer to assert this:
+
+```
+assembly ("memory-safe") { ... }
+```
+
+It can lead to undefined behaviour if you mark an inline assembly block as memory safe if it is
+not actually memory safe, because it can lead to conflicts between stack variables and memory.
+
+The module that can move stack variables to memory is only activated if all inline assembly
+blocks either do not access memory at all or are marked as being memory safe.
+
+If you are e.g. writing a code library that should be used across compiler versions,
+you can also use a special comment syntax that has the same effect:
+``/// @solidity memory-safe-assembly``
+
+We advise against using this syntax, though, as we also advise against using the same
+code base across breaking compiler versions.
```

Comment:
I don't really see the connection to "using the same code base across breaking compiler versions"...
It's neither a breaking release, nor does this have any danger of silent semantics changes one would have in a breaking release...
What we do advice is to always use the latest compiler version - that has more connection to this, doesn't it?

Reply:
My point is, if you are using the comment with `pragma solidity ^0.8.0`, you can just as well use the proper way and require 0.8.13. The main reason for using the comment is that you are also supporting 0.7.x or older.

Reply:
Does that make sense :-)? Anyone supporting both 0.7.x and 0.8.x is rather unlikely due to checked arithmetic alone, isn't it? I was under the impression that the comment was *exactly* meant for supporting ``^0.8.0`` in a library (based on arguments like https://forum.soliditylang.org/t/thoughts-on-override-a-b-syntax/767/14)... if we say anything using ``^0.8.0`` should just jump to 0.8.13, I don't really see the comment version to be of much use after all :-)...

Reply:
Oh yeah that is actually a good point. I mean you could have no arithmetics at all...

Reply:
Looking at the external contracts I had to convert to 0.8.x I remember there being some unaffected by safe math so I don't think having such a library is that unlikely.